src/frontends/pytorch/src/op/im2col.cpp

```diff
@@ -0,0 +1,101 @@
+    auto padding_2 = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {padding_d * 2}));
+    auto stride = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {stride_d}));
+    auto blocks_d = context.mark_node(std::make_shared<opset8::Add>(input_d, padding_2));
+    auto subtrahend =
+        context.mark_node(opset8::Constant::create(element::i64, Shape{}, {dilation_d * (kernel_size_d - 1)}));
+    blocks_d = context.mark_node(std::make_shared<opset8::Subtract>(blocks_d, subtrahend));
+    auto zero = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {0}));
+    auto minus_one = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {-1}));
+    auto blocks_d_indices = context.mark_node(std::make_shared<opset8::Range>(zero, blocks_d, stride, element::i64));
+    blocks_d_indices = context.mark_node(std::make_shared<opset8::Unsqueeze>(blocks_d_indices, zero));
+    std::vector<int64_t> rng;
+    for (int64_t i = 0; i < kernel_size_d * dilation_d; i += dilation_d) {
+        std::cout << i << std::endl;
+        rng.push_back(i);
+    }
+
+    auto kernel_grid = context.mark_node(opset8::Constant::create(element::i64, Shape{rng.size()}, rng));
+    auto kernel_mask = context.mark_node(std::make_shared<opset8::Unsqueeze>(kernel_grid, minus_one));
+    return context.mark_node(std::make_shared<opset8::Add>(blocks_d_indices, kernel_mask));
+}
+
+OutputVector translate_im2col(NodeContext& context) {
+    auto input = context.get_input(0);
+    auto kernel_size = context.const_input<std::vector<int64_t>>(1);
+    auto dilation = context.const_input<std::vector<int64_t>>(2);
+    auto padding = context.const_input<std::vector<int64_t>>(3);
+    auto stride = context.const_input<std::vector<int64_t>>(4);
+    auto input_shape = context.mark_node(std::make_shared<opset8::ShapeOf>(input));
+    auto zero = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {0}));
+    auto zero_f = context.mark_node(opset8::Constant::create(element::f32, Shape{}, {0}));
+    auto minus_one = context.mark_node(opset8::Constant::create(element::i64, Shape{1}, {-1}));
+    auto two = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {2}));
+    auto four = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {4}));
+    auto h_dim = two;
+    auto w_dim = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {3}));
+    auto b_dim = zero;
+    auto c_dim = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {1}));
+    auto input_h = context.mark_node(std::make_shared<opset8::Gather>(input_shape, h_dim, zero));
+    auto input_w = context.mark_node(std::make_shared<opset8::Gather>(input_shape, w_dim, zero));
+    auto input_c = context.mark_node(std::make_shared<opset8::Gather>(input_shape, c_dim, zero));
+    auto input_b = context.mark_node(std::make_shared<opset8::Gather>(input_shape, b_dim, zero));
+    auto stride_h = stride[0];
```

Comment:
Please validate that `stride`, `padding`, `dilation` and `kernal_size` have length of exactly 2.

Reply:
added